core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/data/impl/DataFeedServiceImpl.java

```diff
@@ -301,4 +302,22 @@ private void printViewItemToCsv(CSVPrinter csvPrinter, GridViewItem viewItem) {
     private List<GridResource> doFiltering(List<GridResource> resources, DataFilter filter) {
         return resources.stream().filter(filter::validate).collect(Collectors.toList());
     }
+
+    private static String getUpdatedLink(List<UpdatedItem> updatedItems, String propertyAddress) {
+        return updatedItems.stream()
+                .filter(item -> item.getPropertyLocation().equals(propertyAddress))
+                .map(UpdatedItem::getUpdatedLink)
+                .findFirst()
+                .orElse(StringUtils.EMPTY);
+    }
+
+    private boolean isUpdated(List<UpdatedItem> updatedItems, String propertyAddress, String currentLink) {
+        boolean propertyAddressUpdated = updatedItems.stream()
```

Comment:
Did I get it right, that it's OK that among the `updatedItems` _one item_ can match propertyAddress, and _another item_ can be matching currentLink? Should not they always be _the same_ item? If they shouldn't, skip my comment

Reply:
Also, it is a bit worrying that all the null checks are abandoned. The list is asserted to never be null, and string properties are also asserted to never be null.. is this allright?

Reply:
It worked somehow, but I'm sure it was because of the appropriate cases. Fixed it, thanks!